pubtools/pulplib/_impl/client/client.py

```diff
@@ -179,6 +182,42 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def upload_one_file(
```

Comment:
Not certain currently, but I wonder if it might be better to put the public method onto Repository class instead.

I had similar decision to make for other methods.  For example, to publish a repository you call Repository.publish() instead of Client.publish(repo_id).  Either could work and I don't have a very strong sense of which way is better in general, but since the decision was already made to add publish and delete onto Repository class, I think upload probably makes sense there as well for the sake of consistency.

Reply:
Thinking about it some more, a strong point in favor of Repository.upload is that the library most likely can figure out some things about the upload (at least the type_id and unit_key) based on the type of the Repository.

For example, for a FileRepository, the library knows that type_id must be "iso" and unit_key needs to have name, checksum and size. For a YumRepository, the library knows type_id must be one of ("rpm", "srpm", "yum_repo_metadata_file")... it can verify or maybe even autodetect these things.

Reply:
Okay, I didn't see you put publish in Repository, I was confused since I didn't see how publish is tested, while the test coverage is 100%. 
I guess the name 'model' in model.repository confused me, since I thought it's more like a directory to store database related models, so I didn't check there. 

And yes, it's more reasonable to move the upload to Repository class.